runtime/queries/matlab/highlights.scm

```diff
@@ -1,97 +1,127 @@
+(function_definition
+  "function" @keyword.function
+  name: (identifier) @function
+  [ "end" "endfunction" ]? @keyword.function)
 
-["return" "break" "continue"] @keyword.return
+(function_signature name: (identifier) @function)
+(function_call name: (identifier) @function)
+(handle_operator (identifier) @function)
+(validation_functions (identifier) @function)
+(command (command_name) @function.macro)
+(command_argument) @string
+(return_statement) @keyword.control.return
 
-(
-(identifier) @constant.builtin
-(#any-of? @constant.builtin "true" "false")
-)
+; Assignments
 
-(
-    (identifier) @constant.builtin
-    (#eq? @constant.builtin "end")
-)
+(assignment left: (_) @variable)
+(multioutput_variable (_) @variable)
 
-;; Punctuations
+; Parameters
 
-[";" ","] @punctuation.special
-(argument_list "," @punctuation.delimiter)
-(vector_definition ["," ";"] @punctuation.delimiter)
-(cell_definition ["," ";"] @punctuation.delimiter)
-":" @punctuation.delimiter
-(parameter_list "," @punctuation.delimiter)
-(return_value "," @punctuation.delimiter)
+(function_arguments (identifier) @variable.parameter)
 
-; ;; Brackets
+; Operators
 
 [
- "("
- ")"
- "["
- "]"
- "{"
- "}"
-] @punctuation.bracket
-
-;; Operators
-"=" @operator
-(operation [ ">"
-            "<"
-            "=="
-            "<="
-            ">="
-            "=<"
-            "=>"
-            "~="
-            "*"
-            ".*"
-            "/"
-            "\\"
-            "./"
-            "^"
-            ".^"
-            "+"] @operator)
-
-;; boolean operator
-[
-    "&&"
-    "||"
+  "+"
+  ".+"
+  "-"
+  ".*"
+  "*"
+  ".*"
+  "/"
+  "./"
+  "\\"
+  ".\\"
+  "^"
+  ".^"
+  "'"
+  ".'"
+  "|"
+  "&"
+  "?"
+  "@"
+  "<"
+  "<="
+  ">"
+  ">="
+  "=="
+  "~="
+  "="
+  "&&"
+  "||"
+  ":"
 ] @operator
 
-;; Number
-(number) @constant.numeric
+; Conditionals
+
+(if_statement [ "if" "end" ] @keyword.control.conditional)
+(elseif_clause "elseif" @keyword.control.conditional)
+(else_clause "else" @keyword.control.conditional)
+(switch_statement [ "switch" "end" ] @keyword.control.conditional)
+(case_clause "case" @keyword.control.conditional)
+(otherwise_clause "otherwise" @keyword.control.conditional)
+(break_statement) @keyword.control.conditional
+
+; Repeats
+
+(for_statement [ "for" "parfor" "end" ] @keyword.control.repeat)
+(while_statement [ "while" "end" ] @keyword.control.repeat)
+(continue_statement) @keyword.control.repeat
+
+; Exceptions
 
-;; String
+(try_statement [ "try" "end" ] @keyword.control.exception)
+(catch_clause "catch" @keyword.control.exception)
+
+; Punctuation
+
+[ ";" "," "." ] @punctuation.delimiter
+[ "(" ")" "[" "]" "{" "}" ] @punctuation.bracket
+
+; Literals
+
+(escape_sequence) @escape
+(formatting_sequence) @escape
 (string) @string
+(number) @constant.numeric.float
+(boolean) @constant.builtin.boolean
 
-;; Comment
-(comment) @comment
+; Comments
+
+[ (comment) (line_continuation) ] @comment.line @spell
```

Comment:
The `@spell` capture should probably be removed. I think if we do spell-checking with tree-sitter in core then we'll use a separate query file for the spelling queries than the highlight queries.

Reply:
I saw it was not in the list, but I wasn't sure whether it was used or not. I'll remove it.